react_frontend/src/SimilaritySearch/index.js

```diff
@@ -1,66 +1,77 @@
+import PropTypes from 'prop-types'
 import Router, { useRouter } from 'next/router'
 
 import { getQueryString } from '../Query/helpers'
+import SimilaritySearchThumbnail from './Thumbnail'
 
-const SimilaritySearch = () => {
+const SimilaritySearch = ({ uploadedAssets, setUploadedAssets }) => {
   const {
     pathname,
     query,
-    query: { s },
+    query: { s = '' },
   } = useRouter()
 
-  if (!s) return null
+  if (!s && Object.keys(uploadedAssets).length === 0) return null
 
-  const similarityQuery = s.split(',')
+  const assetIds = s.split(',').filter(Boolean)
 
   return (
     <div className="px-2">
       <h3 className="pt-4 text-sm text-gray-400">
         Showing results similar to:
       </h3>
       <div className="flex py-2 border-b-2 border-gray-700">
-        {similarityQuery.map((assetId) => {
-          const newSimilarityQuery = similarityQuery
+        {Object.entries(uploadedAssets).map(([fileName, dataURL]) => {
+          return (
+            <SimilaritySearchThumbnail
+              key={fileName}
+              assetId={`uploaded asset ${fileName}`}
+              src={dataURL}
+              onClick={() => {
+                const { [fileName]: extract, ...rest } = uploadedAssets
```

Comment:
i don't quite understand this line. you're just renaming the dataURL value belonging to the `filename` asset, right? but why extract that from the assets you pass to `setUploadAssets`?

Reply:
This is a super sneaky technique to remove one key/value pair from an object:
![image](https://user-images.githubusercontent.com/433409/112207100-592b2d80-8bdc-11eb-9005-ae24d07e0445.png)